File: packages/protocol/src/types/deviceRecord.ts
import { z } from "zod";

export const DeviceRecordSchema = z.object({
  type: z.literal("device_record"),
  version: z.literal(1),
  identity_id: z.string(),
  device_id: z.string().uuid(),
  device_pubkey: z.string(),
  device_label: z.string().optional(),
  capabilities: z.object({
    mailbox_poll: z.boolean().optional(),
    receipts: z.boolean().optional(),
    attachments: z.boolean().optional(),
  }),
  created_at_ms: z.number(),
  signature: z.string(),
});

export type DeviceRecord = z.infer<typeof DeviceRecordSchema>;

File: apps/relay/internal/model/device_record.go
package model

type DeviceRecord struct {
	Type         string          `json:"type"`
	Version      int             `json:"version"`
	IdentityID   string          `json:"identity_id"`
	DeviceID     string          `json:"device_id"`
	DevicePubKey string          `json:"device_pubkey"`
	DeviceLabel  *string         `json:"device_label,omitempty"`
	Capabilities map[string]bool `json:"capabilities"`
	CreatedAtMs  int64           `json:"created_at_ms"`
	Signature    string          `json:"signature"`
}
